drivers/adc/ad463x/ad463x.c

```diff
@@ -314,26 +350,50 @@ static int32_t ad463x_init_gpio(struct ad463x_dev *dev,
 				struct ad463x_init_param *init_param)
 {
 	int32_t ret;
-
+	/* configure reset pin */
 	ret = no_os_gpio_get_optional(&dev->gpio_resetn, init_param->gpio_resetn);
 	if (ret != 0)
 		return ret;
 
 	/** Reset to configure pins */
-	if (dev->gpio_resetn) {
-		ret = no_os_gpio_direction_output(dev->gpio_resetn, NO_OS_GPIO_LOW);
-		if (ret != 0)
-			return ret;
+	ret = no_os_gpio_direction_output(dev->gpio_resetn, NO_OS_GPIO_LOW);
+	if (ret != 0)
+		goto error_gpio_reset;
 
-		no_os_mdelay(100);
-		ret = no_os_gpio_set_value(dev->gpio_resetn, NO_OS_GPIO_HIGH);
-		if (ret != 0)
-			return ret;
+	no_os_mdelay(100);
+	ret = no_os_gpio_set_value(dev->gpio_resetn, NO_OS_GPIO_HIGH);
+	if (ret != 0)
+		goto error_gpio_reset;
 
-		no_os_mdelay(100);
+	no_os_mdelay(100);
+
+	/** Configure PGIA pins, if available */
+	if (dev->has_pgia) {
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a0,init_param->gpio_pgia_a0);
+		if (ret != 0)
+			goto error_gpio_reset;
+		ret = no_os_gpio_get_optional(&dev->gpio_pgia_a1, init_param->gpio_pgia_a1);
+		if (ret != 0)
+			goto error_gpio_reset;
+		/* set default values */
+		ret = no_os_gpio_direction_output(dev->gpio_pgia_a0, NO_OS_GPIO_LOW);
+		if (ret != 0)
+			goto error_gpio_pgia;
```

Comment:
in this case dev->dev->gpio_pgia_a1 is not allocated but you are calling remove for it. this will work only if calloc is used (otherwise you will remove some random memory), and I see this driver uses malloc...
I suggest using calloc and then you can use a single label in which you call all remove functions, if the memory is not allocated, it will be null and the remove function will do nothing.